finesse/logger.py

```diff
@@ -8,12 +8,19 @@
 from finesse import config
 
 
-def initialise_logging() -> None:
-    """Configure the program's logger."""
+def get_log_path():
+    """Return the user log path."""
     log_path = user_log_path(config.APP_NAME, config.APP_AUTHOR)
     log_path.mkdir(parents=True, exist_ok=True)
-    filename = log_path / f"{datetime.now().strftime('%Y%m%d_%H-%M-%S')}.log"
+    return log_path
+
+
+log_path = get_log_path()
+filename = log_path / f"{datetime.now().strftime('%Y%m%d_%H-%M-%S')}.log"
```

Comment:
We don't need `log_path` to be a global variable and, now that is used in other places, we need to give `filename` a more descriptive name:
```suggestion
log_file = get_log_path() / f"{datetime.now().strftime('%Y%m%d_%H-%M-%S')}.log"
```

Reply:
I had in mind something like:

```py
"""in finesse/logger.py"""
...

from pathlib import Path

...

log_file: Path

...

def initialise_logging() -> None:
    """Configure the program's logger."""
    log_path = get_log_path()

    global filename
    log_file = log_path / f"{datetime.now().strftime('%Y%m%d_%H-%M-%S')}.log"
...
```

That way the log filename is still defined when logging is initialised rather than when the module is imported